lib/internal/inspector_async_hook.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const { createHook } = require('async_hooks');
+const inspector = process.binding('inspector');
+const errors = require('internal/errors');
+
+if (!inspector || !inspector.asyncTaskScheduled) return;
+
+module.exports = createHook({
+  init(asyncId, type, triggerId, resource) {
+    if (!Number.isInteger(asyncId))
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'asyncId', 'integer');
+    if (typeof type !== 'string')
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'type', 'string');
+    // TODO(bajtos): support recurring tasks like setInterval
```

Comment:
Am I correct to assume that https://v8docs.nodesource.com/node-8.0/df/d37/classv8__inspector_1_1_v8_inspector.html is the most “documentation” we get for the V8 inspector API? If yes, I am not sure how we would want to use it to report recurring tasks… it’s not just setInterval, it’s a lot of things that create multiple events (TCP/UDP servers, file streams, zlib, etc.).